### PR TITLE
[Merged by Bors] - atxs: drop legacy code AwaitATX

### LINKED ISSUE #5015: atxs: subscriptions code is unnecessary
this code is used to notify publisher that atx was saved to database. this is legacy code and has to be removed.

validation for published messages always executed synchronously, so when Publish returns atx will be saved and there is no need to wait for any channel notification.

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation
Closes #5015

### CODE REVIEW

**Comment by countvonzero** [comment]
bors try

File: activation/activation.go [hunk 1]
```diff
@@ -623,19 +617,8 @@ func (b *Builder) PublishActivationTx(ctx context.Context) error {
...
-		}
-	case <-b.layerClock.AwaitLayer((atx.TargetEpoch()).FirstLayer()):
-		if err := b.discardChallenge(); err != nil {
-			return fmt.Errorf("%w: publish epoch has passed", err)
-		}
-		return fmt.Errorf("%w: publish epoch has passed", ErrATXChallengeExpired)
-	case <-ctx.Done():
-		return ctx.Err()
+	if err := b.discardChallenge(); err != nil {
+		return fmt.Errorf("%w: after published atx", err)
```
Comment (poszu) [inline activation/activation.go]: NIT, the error message could be more helpful:
```suggestion
		return fmt.Errorf("discarding challenge after published ATX: %w", err)
```

File: activation/activation_test.go [hunk 2]
```diff
@@ -1047,14 +1000,8 @@ func TestBuilder_NIPostPublishRecovery(t *testing.T) {
 			require.Equal(t, &gotAtx, built)
 			return nil
 		})
-	expireEpoch := publishEpoch + 1
-	tab.mclock.EXPECT().AwaitLayer(expireEpoch.FirstLayer()).Return(done)
-	tab.mhdlr.EXPECT().UnsubscribeAtx(gomock.Any()).Do(
-		func(got types.ATXID) {
-			require.Equal(t, built.ID(), got)
-		})
 	// This 👇 ensures that handing of the challenge succeeded and the code moved on to the next part
```
Comment (poszu) [inline activation/activation_test.go]: I think this comment is no longer valid and necessary.

**Review by poszu — APPROVED** [review]

**Comment by poszu** [comment]
weird env error in CI. bors try

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**